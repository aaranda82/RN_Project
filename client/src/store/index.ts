import {Action, action, createStore, createTypedHooks} from 'easy-peasy';

export interface StoreModel {
  userId: string;
  setUserId: Action<StoreModel, string>;
  clearUserId: Action<StoreModel>;
}

export const store = createStore<StoreModel>({
  userId: '',
  setUserId: action((state, payload) => {
    state.userId = payload;
  }),
  clearUserId: action(state => {
    state.userId = '';
  }),
});

const typedHooks = createTypedHooks<StoreModel>();

export const useStoreActions = typedHooks.useStoreActions;
export const useStoreDispatch = typedHooks.useStoreDispatch;
export const useStoreState = typedHooks.useStoreState;
