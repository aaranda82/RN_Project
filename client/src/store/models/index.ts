import { userModel, UserModel } from './user';

export interface StoreModel {
  user: UserModel;
}

export const storeModel: StoreModel = {
  user: userModel,
};
