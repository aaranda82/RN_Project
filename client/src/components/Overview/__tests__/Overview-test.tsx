import React from 'react';
import OverView from '../Overview';

import renderer from 'react-test-renderer';
import { createStore, StoreProvider } from 'easy-peasy';
import { storeModel } from '../../../store';

const navProps: any = {};
const routeProps: any = {};

it('renders RegisterForm correctly', () => {
  const store = createStore(storeModel);
  const tree = renderer
    .create(
      <StoreProvider store={store}>
        <OverView navigation={navProps} route={routeProps} />
      </StoreProvider>,
    )

    .toJSON();
  expect(tree).toMatchSnapshot();
});
