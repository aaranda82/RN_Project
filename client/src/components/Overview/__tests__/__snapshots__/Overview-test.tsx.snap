// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders RegisterForm correctly 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 40,
        "marginBottom": 40,
        "marginTop": 40,
        "textAlign": "center",
      }
    }
  >
    Hi !
  </Text>
  <Text
    style={
      Object {
        "fontSize": 40,
        "marginBottom": 40,
        "marginTop": 40,
        "textAlign": "center",
      }
    }
  >
    You have logged in
  </Text>
  <Text
    style={
      Object {
        "fontSize": 40,
        "marginBottom": 40,
        "marginTop": 40,
        "textAlign": "center",
      }
    }
  >
    
  </Text>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Am I logged in?
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Log out
      </Text>
    </View>
  </View>
</View>
`;
